### PR TITLE
[GH-669] Channels compatiblity tests

### LINKED ISSUE #669: Channels compatibility tests
The big channels PR currently doesn't contain compatibility tests, but is ready to be merged. To not delay it further let's add tests separately.

### PR DESCRIPTION
**Pull request by cytadela8**
I have made some time ago Channels compatibility tests. They were not merged, because they were not passing. They started passing after the recent ChannelOffChainTx serialization fixes. To not to add more code into the big #586 PR, I separate this changes.

This branch is rebased GH-543. That contained my original work.

This also includes two very small fixes to channels.

This is blocked till #586 is merged, becouse it's on top of that PR.

This closes #669

### CODE REVIEW

File: apps/aecore/test/aecore_channels_compatibility_test.exs [hunk 1]
```diff
@@ -197,28 +199,217 @@ defmodule AecoreChannelCompatibilityTest do
...
+
+    assert_channels_equal(our_channel, epoch_channel)
+
+    assert :ok == call_s1({:receive_confirmed_tx, open_tx})
+    assert :ok == call_s2({:receive_confirmed_tx, open_tx})
+
+    # transfer from responder to initiator with amount 70
+    epoch_transfer1 =
+      Bits.decode58(
+        "G6DYbwbT9jsa4M7dKjQAw1bSwcPN3TcaCKtxBBAk28QaXfEugae4HGVNxSHSrSd59vPCLAssWmX8vN48LWwTypFq2FYy8AVAev68Q5WerLhcsCViNAtgNdzdeGhRMYHBHxYwnPdMRjKThcQAogdqjrtnMKnwqVE6awwDPYPmRjsDT3JW8HY8m43SWB9QeiauG3EZx8KnjejyXuNZNPGVjFiGnjQCAhwVb5nPXuhs4KXgTKjB5w1Huch5wJg3HUzyQXhuoCk4gn7vcQvjX7P4rthpo1CMNfJj6WXFowKpVzUCQrVrTYUMHzvbYuymzh1CdcmfgJouxHQmDSHXJwRW6Uw6uwRY192oKMbhsanuLjRJC9YB8SzSLdrSu828c6QLHnWqPedsHwW"
```
Comment (thepiwo) [inline apps/aecore/test/aecore_channels_compatibility_test.exs]: I assume most of the hardcoded things will break with all updates to the protocol?

File: apps/aecore/lib/aecore/channel/channel_state_on_chain.ex [hunk 2]
```diff
@@ -308,7 +308,8 @@ defmodule Aecore.Channel.ChannelStateOnChain do
       | slash_close: block_height + channel.lock_period,
         slash_sequence: offchain_tx.sequence,
```
[the remark this answers is not in the document]
Comment (cytadela8) [reply apps/aecore/lib/aecore/channel/channel_state_on_chain.ex]: fixed

**Comment by thepiwo** [comment]
@cytadela8 should I merge?

**Comment by cytadela8** [comment]
I think we can merge.

**Pull request merged**